frontend/app/src/main/java/com/quickpick/MainActivity.java

```diff
@@ -31,44 +45,48 @@ private void registerButtons() {
     }
 
     private void setOnClickListeners() {
-        createNewList.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                startActivity(new Intent(getApplicationContext(), CreateNewListActivity.class));
-            }
-        });
+        createNewList.setOnClickListener(view ->
+                startActivity(new Intent(getApplicationContext(), CreateNewListActivity.class))
+        );
 
-        viewEditLists.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                startActivity(new Intent(getApplicationContext(), ViewEditListsActivity.class));
-            }
-        });
+        viewEditLists.setOnClickListener(view ->
+                startActivity(new Intent(getApplicationContext(), ViewEditListsActivity.class))
+        );
 
-        joinSession.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                new AlertDialog.Builder(MainActivity.this)
-                        .setMessage(getString(R.string.enter_session_id_title))
-                        .setTitle(getString(R.string.enter_session_id_title))
-                        .create()
-                        .show();
-            }
-        });
+        joinSession.setOnClickListener(view -> showAlertDialog());
 
-        viewOldSessions.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                startActivity(new Intent(getBaseContext(), ViewOldSessionsActivity.class));
-            }
-        });
+        viewOldSessions.setOnClickListener(view ->
+                startActivity(new Intent(getBaseContext(), ViewOldSessionsActivity.class))
+        );
 
-        createSession.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                startActivity(new Intent(getBaseContext(), SessionActivity.class));
-            }
+        createSession.setOnClickListener(view -> {
+            SessionPayload payload = new SessionPayload();
+            navigateToSessionActivity(payload);
+            // TODO: Update this to do a POST to the server to create the session
         });
+    }
+
+    private void showAlertDialog() {
+        AlertDialog dialog = new AlertDialog.Builder(MainActivity.this)
+                .setTitle(getString(R.string.enter_session_id_title))
+                .setView(R.layout.dialog_join_session)
+                .create();
+        dialog.show();
+        EditText sessionCode = dialog.findViewById(R.id.session_code_edit_text);
```

Comment:
maybe some whitespace here? kind of chunky

Reply:
Added some whitespace